src/lib.rs

```diff
@@ -1597,12 +1640,6 @@ where
     I: Input<'a>,
     E: ParserExtra<'a, I>,
 {
-    // TODO: Unfortunately we can't constrain the trait above without a cycle. Figure out a way to have this associated
```

Comment:
The type is already usable from external crates a `P::Config` - sealed traits associated types can still be accessed (may want to make sure this doesn't cause any issues)

Reply:
Ah, nice. That's good to know.

Unfortunately, it doesn't appear in docs :(

Reply:
Unfortunate. May be worth adding a section on it at some point